grakn-engine/src/main/java/ai/grakn/engine/postprocessing/PostProcessing.java

```diff
@@ -173,37 +134,22 @@ private void waitToContinue() {
         futures.clear();
     }
 
-    private void dumpStats() {
-        while (isRunning.get()) {
-            LOG.info("--------------------Current Status of Post Processing--------------------");
-            dumpStatsType("Casting");
-            dumpStatsType("Resources");
-            LOG.info("Current Stage: " + currentStage);
-            LOG.info("-------------------------------------------------------------------------");
-
-            try {
-                Thread.sleep(30000);
-            } catch (InterruptedException e) {
-                LOG.error("Exception",e);
-            }
-        }
+    private void dumpStats(String keyspace, String type, Set<ConceptId> concepts) {
+        LOG.info("--------------------Current Status of Post Processing--------------------");
+        LOG.info("Keyspace      : " + keyspace);
+        LOG.info("" + type + "      : " + concepts.size());
+        LOG.info("-------------------------------------------------------------------------");
     }
 
-    private void dumpStatsType(String typeName) {
-        long total = 0L;
-        LOG.info(typeName + " Jobs:");
-
-        for (String keyspace : cache.getKeyspaces()) {
-            long numJobs = 0L;
-            if(typeName.equals("Casting")){
-                numJobs = cache.getNumCastingJobs(keyspace);
-            } else if(typeName.equals("Resources")){
-                numJobs = cache.getNumResourceJobs(keyspace);
-            }
-            LOG.info("        For Graph [" + keyspace + "] has jobs : " + numJobs);
-            total += numJobs;
-        }
-
-        LOG.info("    Total Jobs: " + total);
+    /**
+     * Functional interface for the perform post processing methods
+     * @param <A> the keyspace
+     * @param <B> the index of the concept
+     * @param <C> the set of concept ids to post process
+     */
+    @FunctionalInterface
```

Comment:
We should delete this. (I have code that already does, which I can submit as a PR once this is in central)